payload-schemas/schemas/fork/event.schema.json

```diff
@@ -5,185 +5,10 @@
-        "compare_url": { "type": "string" },
-        "merges_url": { "type": "string" },
-        "archive_url": { "type": "string" },
-        "downloads_url": { "type": "string" },
-        "issues_url": { "type": "string" },
-        "pulls_url": { "type": "string" },
-        "milestones_url": { "type": "string" },
-        "notifications_url": { "type": "string" },
-        "labels_url": { "type": "string" },
-        "releases_url": { "type": "string" },
-        "deployments_url": { "type": "string" },
-        "created_at": {
-          "oneOf": [{ "type": "integer" }, { "type": "string" }]
-        },
-        "updated_at": { "type": "string" },
-        "pushed_at": { "oneOf": [{ "type": "integer" }, { "type": "string" }] },
-        "git_url": { "type": "string" },
-        "ssh_url": { "type": "string" },
-        "clone_url": { "type": "string" },
-        "svn_url": { "type": "string" },
-        "homepage": { "type": ["string", "null"] },
-        "size": { "type": "integer" },
-        "stargazers_count": { "type": "integer" },
-        "watchers_count": { "type": "integer" },
-        "language": { "type": ["string", "null"] },
-        "has_issues": { "type": "boolean" },
-        "has_projects": { "type": "boolean" },
-        "has_downloads": { "type": "boolean" },
-        "has_wiki": { "type": "boolean" },
-        "has_pages": { "type": "boolean" },
-        "forks_count": { "type": "integer" },
-        "mirror_url": { "type": ["string", "null"] },
-        "archived": { "type": "boolean" },
-        "disabled": { "type": "boolean" },
-        "open_issues_count": { "type": "integer" },
-        "license": {
-          "oneOf": [
-            { "$ref": "common/license.schema.json" },
-            { "type": "string" },
-            { "type": "null" }
-          ]
-        },
-        "forks": { "type": "integer" },
-        "open_issues": { "type": "integer" },
-        "watchers": { "type": "integer" },
-        "stargazers": { "type": "integer" },
-        "default_branch": { "type": "string" },
-        "allow_squash_merge": { "type": "boolean" },
-        "allow_merge_commit": { "type": "boolean" },
-        "allow_rebase_merge": { "type": "boolean" },
-        "delete_branch_on_merge": { "type": "boolean" },
-        "master_branch": { "type": "string" },
-        "permissions": {
-          "type": "object",
-          "required": ["pull", "push", "admin"],
-          "properties": {
-            "pull": { "type": "boolean" },
-            "push": { "type": "boolean" },
-            "admin": { "type": "boolean" },
-            "maintain": { "type": "boolean" },
-            "triage": { "type": "boolean" }
-          },
-          "additionalProperties": false
-        },
-        "public": { "type": "boolean" }
-      },
-      "additionalProperties": false
+      "allOf": [
+        { "$ref": "common/repository.schema.json" },
+        { "type": "object", "properties": { "fork": { "enum": [true] } } }
```

Comment:
You will need to add `"tsAdditionalProperties": false` to the second schema object in the `allOf`

Reply:
Done. :heavy_check_mark:

It seems annoying that json-schema-to-typescript doesn't automatically exclude members that are outside the intersection, without needing `"additionalProperties": false`?

Reply:
That's because of how JSON schemas work. I'll refer you to @G-Rath he came up with the solution as to how to "extend" the common schemas in #331
We extended the spec and are using that custom property to tell the compiler to not allow other properties.

Reply:
Thanks for the pointer to #331: does https://github.com/jablko/webhooks/blob/7a95e2474b58109bf8fc942d1530e9927e3510e2/schema.d.ts#L1795 make the type an open object? I assume the intersection of an open object and a closed object is a closed object, so `"tsAdditionalProperties": false`/suppressing `[k: string]: unknown` is cosmetic, not semantic?

Reply:
Its not just cosmetic. It has semantic value as well.

We use it the same as the `additionalProperties` meta key in JSON schema spec when we don't want additional properties. Its use is only for "extending" schemas.

It tells the test suite and the compiler that it includes the other schema, and that it replaces the properties contained with the more specific values in the second schema, and not to allow additional properties

Reply:
#331 says `tsAdditionalProperties` is only used when compiling a schema to TypeScript, and I could be wrong, but I assume e.g. `{ fork: boolean } & { fork: true }` and `{ fork: boolean } & { fork: true, [k: string]: unknown }` are semantically equivalent (both closed objects)?

Reply:
They are not: the former says there is only one property (`fork`), the other says there is *at least* `fork` but  could be other properties whose types we don't know.

This has very subtle implications in TypeScript relating to things like passing in computed properties and the compatibility of defined properties (as they all must match their index signature).

It also can degrade IDE autocompletion since they tend to suggest different things based on the fact that it's an open object.